Docs: Document the Visual Testing addon

jonniebigodes opened the pull request:
<!-- If your PR is related to an issue, provide the number(s) above; if it resolves multiple issues, be sure to break them up (e.g. "closes #1000, closes #1001"). -->

<!--

Thank you for contributing to Storybook! Please submit all PRs to the `next` branch unless they are specific to the current release. Storybook maintainers cherry-pick bug and documentation fixes into the `main` branch as part of the release process, so you shouldn't need to worry about this. For additional guidance: https://storybook.js.org/docs/contribute

-->

## What I did

With this pull request, the visual testing documentation was updated to factor in the Visual tests addon.

What was done:
- Removed the previous assets as they were no longer relevant
- Updated the docs to include instructions of how the addon is set up and also the workflows
- Created the required snippets


@kylegach I've assigned this to you so that you can take it from here and provide the required changes. Also @domyen if you could assist on some of the designs/animations would appreciate it.


 ## Checklist for Contributors

### Testing

1. Follow the steps in the [contributing instructions](https://storybook.js.org/docs/react/contribute/new-snippets#preview-your-work) for this branch, `wip_docs_vt_addon`.
2. Open the relevant documentation and verify the feature changes.

#### The changes in this PR are covered in the following automated tests:
- [ ] stories
- [ ] unit tests
- [ ] integration tests
- [ ] end-to-end tests

#### Manual testing

_This section is mandatory for all contributions. If you believe no manual test is necessary, please state so explicitly. Thanks!_

<!-- Please include the steps to test your changes here. For example:

1. Run a sandbox for template, e.g. `yarn task --task sandbox --start-from auto --template react-vite/default-ts`
2. Open Storybook in your browser
3. Access X story

-->

### Documentation

<!-- Please check (put an "x" inside the "[ ]") the applicable items below to indicate which documentation has been updated. -->

- [ ] Add or update documentation reflecting your changes
- [ ] If you are deprecating/removing a feature, make sure to update
      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

## Checklist for Maintainers

- [ ] When this PR is ready for testing, make sure to add `ci:normal`, `ci:merged` or `ci:daily` GH label to it to run a specific set of sandboxes. The particular set of sandboxes can be found in `code/lib/cli/src/sandbox-templates.ts`
- [ ] Make sure this PR contains **one** of the labels below:
   <details>
     <summary>Available labels</summary>

     - `bug`: Internal changes that fixes incorrect behavior.
     - `maintenance`: User-facing maintenance tasks.
     - `dependencies`: Upgrading (sometimes downgrading) dependencies.
     - `build`: Internal-facing build tooling & test updates. Will not show up in release changelog.
     - `cleanup`: Minor cleanup style change. Will not show up in release changelog.
     - `documentation`: Documentation **only** changes. Will not show up in release changelog.
     - `feature request`: Introducing a new feature.
     - `BREAKING CHANGE`: Changes that break compatibility in some way with current major version.
     - `other`: Changes that don't fit in the above categories.
   
   </details>

### 🦋 Canary release

<!-- CANARY_RELEASE_SECTION -->

This PR does not have a canary release associated. You can request a canary release of this pull request by mentioning the `@storybookjs/core` team here.

_core team members can create a canary release [here](https://github.com/storybookjs/storybook/actions/workflows/canary-release-pr.yml) or locally with `gh workflow run --repo storybookjs/storybook canary-release-pr.yml --field pr=<PR_NUMBER>`_

<!-- CANARY_RELEASE_SECTION -->

jonniebigodes committed:
Docs: Visual testing addon updates

jonniebigodes commented on docs/snippets/common/visual-test-addon-install.npx.js.mdx:
````diff
@@ -0,0 +1,3 @@
+```shell
+npx storybook@latest add @chromaui/addon-visual-tests
````
@kylegach, heads up: when addressing these snippets, they shouldn't be suffixed with `next` as this is a satellite repository and doesn't contain a `next` branch that allows users to install pre-release versions similar to other addons.

kylegach replied:
I double-checked the logic; only `@storybook/...` packages are affected.

jonniebigodes replied:
Appreciate you taking the time to check this, glad that we don't need to do anything extra.

Commit:
Update VT explanatory animation to remove visual artifact

Commit:
Pass at Visual Test docs

Commit:
Update visual tests page with new images & CI instructions

domyen commented:
@kylegach I pushed the changes to the prose and images. Can I handball this to you?

kylegach committed:
Content updates

- Hide the RendererSelector, because the content doesn't depend on the renderer
- Improve consistency of snippet filenames
- Required properties listed first in table
- Fixed heading levels
- Small prose tweaks, mostly around consistency with rest of docs

kylegach commented on docs/writing-tests/visual-testing.md:
```diff
@@ -13,76 +14,113 @@ Ideal for verifying what the user sees: layout, color, size, and contrast. Story
 ---
 
-#### What’s the difference between visual tests and snapshot tests?
+### What’s the difference between visual tests and snapshot tests?
```
@domyen — I realize you made this an H4 so that it had less visual prominence. But now that we have the "On this page" TOC, the incorrect heading level made this appear to be under the CI section. I viewed that confusion as worse than the visual aspect.

Also, I added a brief paragraph on visual tests, to more completely answer the actual question here. Feel free to tweak!

domyen replied:
Thanks for the explanation about the H4! 

Your paragraph looks great ❤️

kylegach committed:
Update package name
